### DISCUSSION TITLE
Where to control the delete item button in repeater to ask for confirmation and make some calculations?

### DISCUSSION THREAD

**Discussion by albertlens** [filamentphp/filament#6097] [Help]
Hello.

First of all I want to say that I am very happy and satisfied using FilamentPHP as the best TALL solution I have found.

I have a DeliveryNotes Resource where user creates a new Delivery Note and adds ITEMS (services - products). I am using a Repeater::make('items')->relationship('items')... and everything is working fine except when user deletes the last item, and I do not know where to control it to ask for confirmation, to prevent deleting of the last item or, even better, to make some TOTAL AMOUNT (delivery note total amount) calculation to zero.

Any ideas, please?

Thanks in advance to everybody.

**Comment by danharrin** [comment]
The confirmation modal is possible in v3, and the calculation can be done inside an `afterStateUpdated()` on a field, or a `mutateDataBeforeCreate()` etc hook on an admin panel page, or a model observer

**Reply by albertlens** [reply]
Hi and thank you.
I am already using afterStateUpdated() and even a Placehoder with the content function method to keep and mantain the delivery note total amount updated with the sum of the items' amount. But the problem is that if user deletes the last item with the delete button, it does nothing about the calculation (it does not trigger the afterStateUpdated() method or even the Placeholder ->content(function ) does not update the value).
For example, if I have one only item with an amount of 120Euros, the DeliveryNote total amount says also 120Euros (which is ok). But if I delete that only item, the DeliveryNote total amount still says 120Euros instead of ZERO.

By the way, I understand version 3 is not available yet. When will it be realeased?

Thanks anyhow for your help.

**Reply by danharrin** [reply]
`content(fn () => ...)` should refresh when you remove a repeater item, as it is "stateless" and I shouldn't hold a value that needs to be set from `afterStateUpdated()`.

So I would use `content()` only, not `afterStateUpdated()`

**Reply by albertlens** [reply]
Yes, with:

->visible(fn (Get $get) => !$get('record_type_id') == null && !$get('opening_date') == null)
or ->visible(true)

you can use true or false or a function which returns a true or false boolean.

But better go to see the documents in the official website:
https://filamentphp.com/docs
